mesh/mesh-io-tcpserver.c

```diff
@@ -203,19 +233,28 @@ static void io_error_callback(struct silvair_io *silvair_io)
 	}
 }
 
+static bool match_tcpserver_conn_by_silvair_io(const void *a, const void *b)
+{
+	const struct tcpserver_conn *conn = a;
+	const struct silvair_io *io = b;
+
+	return conn->io == io;
+}
+
 static void io_disconnect_callback(struct silvair_io *silvair_io)
 {
 	struct mesh_io *mesh_io = silvair_io->context;
+	struct tcpserver_conn *conn;
 
-	if (!mesh_io->pvt->client_io)
+	if (!mesh_io->pvt->conns)
 		return;
 
-	if (!l_queue_remove(mesh_io->pvt->client_io, silvair_io)) {
-		perror("l_queue_remove() error");
-		abort();
-	}
+	conn = l_queue_find(mesh_io->pvt->conns,
+				match_tcpserver_conn_by_silvair_io, silvair_io);
+	if (conn)
+		conn->io = NULL;
```

Comment:
Shouldn't we `conn->conn_stat->connected = false` here?

Also, this is the point where you should emit `PropertiesChanged` signal.

Reply:
`conn_stat` will provide functions that will emit signals (as we discussed). I've put it here just for test purpose.